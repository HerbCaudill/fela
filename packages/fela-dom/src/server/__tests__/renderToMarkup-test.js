import { createRenderer } from 'fela'
import { html as beautify } from 'js-beautify'

import renderToMarkup from '../renderToMarkup'

describe('Rendering to HTML markup', () => {
  it('should return a single HTML markup string', () => {
    const rule = props => ({
      color: props.color,
      '@supports (display:flex)': {
        color: 'yellow',
      },
      '@supports (display:grid)': {
        color: 'brown',
      },
      '@media (min-height: 300px)': {
        color: 'blue',
        '@supports (display:flex)': {
          color: 'green',
        },
        '@supports (display:grid)': {
          color: 'black',
        },
      },
    })

    const renderer = createRenderer()
    renderer.renderRule(rule, {
      color: 'red',
    })
    renderer.renderStatic('*{box-sizing:border-box}')
    renderer.renderStatic(
      {
        display: 'flex',
      },
      'div'
    )

    expect(beautify(renderToMarkup(renderer))).toMatchSnapshot()
  })

  it('should return a single HTML markup string', () => {
    const rule = () => ({
      color: 'yellow',
      ':hover': {
        color: 'blue',
      },
      ':focus-within': {
        color: 'blue',
      },
      ':active': {
        color: 'red',
      },
      '@media (min-width: 300px)': {
        color: 'darkgrey',
        ':active': {
          color: 'blue',
        },
        ':hover': {
          color: 'red',
        },
        ':focus-within': {
          color: 'purple',
        },
      },
    })

    const rule2 = () => ({
      color: 'grey',
      ':hover': {
        color: 'black',
      },
      ':focus-within': {
        color: 'brown',
      },
      ':active': {
        color: 'white',
      },
    })

    const renderer = createRenderer()

    renderer.renderRule(rule)
    renderer.renderRule(rule2)

    expect(beautify(renderToMarkup(renderer))).toMatchSnapshot()
  })
})
