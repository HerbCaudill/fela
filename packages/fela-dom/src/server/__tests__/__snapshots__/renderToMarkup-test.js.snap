// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering to HTML markup should return a single HTML markup string 1`] = `
"<style type=\\"text/css\\" data-fela-rehydration=\\"6\\" data-fela-type=\\"STATIC\\">
    * {
        box-sizing: border-box
    }

    div {
        display: flex
    }
</style>
<style type=\\"text/css\\" data-fela-rehydration=\\"6\\" data-fela-type=\\"RULE\\">
    .a {
        color: red
    }
</style>
<style type=\\"text/css\\" data-fela-rehydration=\\"6\\" data-fela-type=\\"RULE\\" data-fela-support=\\"true\\">
    @supports (display:flex) {
        .b {
            color: yellow
        }
    }

    @supports (display:grid) {
        .c {
            color: brown
        }
    }
</style>
<style type=\\"text/css\\" data-fela-rehydration=\\"6\\" data-fela-type=\\"RULE\\" media=\\"(min-height: 300px)\\">
    .d {
        color: blue
    }
</style>
<style type=\\"text/css\\" data-fela-rehydration=\\"6\\" data-fela-type=\\"RULE\\" data-fela-support=\\"true\\" media=\\"(min-height: 300px)\\">
    @supports (display:flex) {
        .e {
            color: green
        }
    }

    @supports (display:grid) {
        .f {
            color: black
        }
    }
</style>"
`;

exports[`Rendering to HTML markup should return a single HTML markup string 2`] = `
"<style type=\\"text/css\\" data-fela-rehydration=\\"12\\" data-fela-type=\\"RULE\\">
    .a {
        color: yellow
    }

    .i {
        color: grey
    }

    .j:hover {
        color: black
    }

    .b:hover {
        color: blue
    }

    .c:focus-within {
        color: blue
    }

    .k:focus-within {
        color: brown
    }

    .d:active {
        color: red
    }

    .l:active {
        color: white
    }
</style>
<style type=\\"text/css\\" data-fela-rehydration=\\"12\\" data-fela-type=\\"RULE\\" media=\\"(min-width: 300px)\\">
    .e {
        color: darkgrey
    }

    .g:hover {
        color: red
    }

    .h:focus-within {
        color: purple
    }

    .f:active {
        color: blue
    }
</style>"
`;
